app/resources/v1/geoip.js

```diff
@@ -0,0 +1,101 @@
+const config = require('config')
+const request = require('request')
+const logger = require('../../../lib/logger.js')()
+const redis = require('redis')
+const url = require('url')
+
+const IP_GEOLOCATION_TIMEOUT = 500
+
+exports.get = function (req, res) {
+  if (req.headers.host !== config.app_host_port) {
+    res.status(403).json({ status: 403, error: 'Not allowed to access API' })
+    return
+  }
+
+  const requestGeolocation = function (isRedisConnected = true) {
+    console.log('requesting geolocation')
+    const url = `${config.geoip.protocol}${config.geoip.host}?access_key=${config.geoip.api_key}`
+
+    request(url, { timeout: IP_GEOLOCATION_TIMEOUT }, function (error, response, body) {
```

Comment:
Does this request ever pass in the IP address we're trying to look up?

Reply:
No, the endpoint detects the ip address which the current API is coming from ([documentation](https://ipstack.com/documentation#requester)). It returns the ip address in the response.